source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -0,0 +1,132 @@
+      // of trying to map it into a 64-bit space.
+      stream_info_(dispatcher.timeSource()), id_(quic_connection_->connection_id().Hash()) {
+  // TODO(danzh): Use QUIC specific enum value.
+  stream_info_.protocol(Http::Protocol::Http2);
+}
+
+void QuicFilterManagerConnectionImpl::addWriteFilter(Network::WriteFilterSharedPtr filter) {
+  filter_manager_.addWriteFilter(filter);
+}
+
+void QuicFilterManagerConnectionImpl::addFilter(Network::FilterSharedPtr filter) {
+  filter_manager_.addFilter(filter);
+}
+
+void QuicFilterManagerConnectionImpl::addReadFilter(Network::ReadFilterSharedPtr filter) {
+  filter_manager_.addReadFilter(filter);
+}
+
+bool QuicFilterManagerConnectionImpl::initializeReadFilters() {
+  return filter_manager_.initializeReadFilters();
+}
+
+void QuicFilterManagerConnectionImpl::addConnectionCallbacks(Network::ConnectionCallbacks& cb) {
+  network_connection_callbacks_.push_back(&cb);
+}
+
+void QuicFilterManagerConnectionImpl::addBytesSentCallback(
+    Network::Connection::BytesSentCb /*cb*/) {
+  // TODO(danzh): implement to support proxy. This interface is only called from
+  // TCP proxy code.
+  ASSERT(false, "addBytesSentCallback is not implemented for QUIC");
+}
+
+void QuicFilterManagerConnectionImpl::enableHalfClose(bool enabled) {
+  ASSERT(!enabled, "Quic connection doesn't support half close.");
+}
+
+void QuicFilterManagerConnectionImpl::setBufferLimits(uint32_t /*limit*/) {
+  // TODO(danzh): add interface to quic for connection level buffer throttling.
+  // Currently read buffer is capped by connection level flow control. And
+  // write buffer is not capped.
+  ENVOY_CONN_LOG(error, "Quic manages its own buffer currently.", *this);
+}
+
+uint32_t QuicFilterManagerConnectionImpl::bufferLimit() const {
+  // As quic connection is not HTTP1.1, this method shouldn't be called by HCM.
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+void QuicFilterManagerConnectionImpl::close(Network::ConnectionCloseType type) {
+  if (type != Network::ConnectionCloseType::NoFlush) {
+    // TODO(danzh): Implement FlushWrite and FlushWriteAndDelay mode.
+    ENVOY_CONN_LOG(error, "Flush write is not implemented for QUIC.", *this);
+  }
+  quic_connection_->CloseConnection(quic::QUIC_NO_ERROR, "Closed by application",
+                                    quic::ConnectionCloseBehavior::SEND_CONNECTION_CLOSE_PACKET);
+}
+
+void QuicFilterManagerConnectionImpl::setDelayedCloseTimeout(std::chrono::milliseconds timeout) {
+  ASSERT(timeout == std::chrono::milliseconds::zero(),
+         "Delayed close of connection is not supported");
+}
+
+std::chrono::milliseconds QuicFilterManagerConnectionImpl::delayedCloseTimeout() const {
+  // Not called outside of Network::ConnectionImpl. Maybe remove this interface
+  // from Network::Connection.
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+const Network::ConnectionSocket::OptionsSharedPtr&
+QuicFilterManagerConnectionImpl::socketOptions() const {
+  ENVOY_CONN_LOG(
+      error,
```

Comment:
Per elsewhere, let's please not log at error, etc. Let's either implement or NOT_IMPLEMENTED everywhere in this file.